test/extensions/clusters/aggregate/cluster_integration_test.cc

```diff
@@ -0,0 +1,285 @@
+        - cluster_2
+  listeners:
+    name: http
+    address:
+      socket_address:
+        address: 127.0.0.1
+        port_value: 0
+    filter_chains:
+      filters:
+        name: envoy.http_connection_manager
+        config:
+          stat_prefix: config_test
+          http_filters:
+            name: envoy.router
+          codec_type: HTTP2
+          route_config:
+            name: route_config_0
+            validate_clusters: false
+            virtual_hosts:
+              name: integration
+              routes:
+              - route:
+                  cluster: cluster_1
+                match:
+                  prefix: "/cluster1"
+              - route:
+                  cluster: cluster_2
+                match:
+                  prefix: "/cluster2"
+              - route:
+                  cluster: aggregate_cluster
+                match:
+                  prefix: "/aggregatecluster"
+              domains: "*"
+)EOF",
+      api_type);
+}
+
+class AggregateIntegrationTest : public Grpc::DeltaSotwIntegrationParamTest,
+                                 public HttpIntegrationTest {
+public:
+  AggregateIntegrationTest()
+      : HttpIntegrationTest(
+            Http::CodecClient::Type::HTTP2, ipVersion(),
+            config(sotwOrDelta() == Grpc::SotwOrDelta::Sotw ? "GRPC" : "DELTA_GRPC")) {
+    use_lds_ = false;
+    sotw_or_delta_ = sotwOrDelta();
+  }
+
+  void TearDown() override {
+    if (!test_skipped_) {
+      cleanUpXdsConnection();
+      test_server_.reset();
+      fake_upstreams_.clear();
+    }
+  }
+
+  // Overridden to insert this stuff into the initialize() at the very beginning of
+  // HttpIntegrationTest::testRouterHeaderOnlyRequestAndResponse().
+  void initialize() override {
+    use_lds_ = false;
+    test_skipped_ = false;
+    // Controls how many fake_upstreams_.emplace_back(new FakeUpstream) will happen in
+    // BaseIntegrationTest::createUpstreams() (which is part of initialize()).
+    // Make sure this number matches the size of the 'clusters' repeated field in the bootstrap
+    // config that you use!
+    setUpstreamCount(2);                                  // the CDS cluster
+    setUpstreamProtocol(FakeHttpConnection::Type::HTTP2); // CDS uses gRPC uses HTTP2.
+
+    // HttpIntegrationTest::initialize() does many things:
+    // 1) It appends to fake_upstreams_ as many as you asked for via setUpstreamCount().
+    // 2) It updates your bootstrap config with the ports your fake upstreams are actually listening
+    //    on (since you're supposed to leave them as 0).
+    // 3) It creates and starts an IntegrationTestServer - the thing that wraps the almost-actual
+    //    Envoy used in the tests.
+    // 4) Bringing up the server usually entails waiting to ensure that any listeners specified in
+    //    the bootstrap config have come up, and registering them in a port map (see lookupPort()).
+    //    However, this test needs to defer all of that to later.
```

Comment:
does this test need this? or is this just copied from another test?

Reply:
Yeah, I c+v here. I will remove unrelated comments.